tests/test_requests.py

```diff
@@ -482,6 +482,51 @@ def test_POSTBIN_GET_POST_FILES(self, httpbin):
         with pytest.raises(ValueError):
             requests.post(url, files=['bad file data'])
 
+    def test_POSTBIN_SEEKED_OBJECT_WITH_NO_ITER(self, httpbin):
+
+        class BufferedStream(object):
+            def __init__(self, data):
+                if isinstance(data, buffer):
+                    self.data = data
+                else:
+                    self.data = buffer(data)
```

Comment:
This is over-general I think. We only ever pass one type here, so the if branch is unnecessary.